api/geth_backend.go

```diff
@@ -2859,3 +2853,29 @@ func (b *GethStatusBackend) TogglePanicReporting(enabled bool) error {
 	}
 	return b.DisablePanicReporting()
 }
+
+func (b *GethStatusBackend) SetLogLevel(level string) error {
+	b.mu.Lock()
+	defer b.mu.Unlock()
+
+	err := nodecfg.SetLogLevel(b.appDB, level)
+	if err != nil {
+		return err
+	}
+	b.config.LogLevel = level
+
+	return logutils.OverrideRootLoggerWithConfig(b.config.LogSettings())
+}
+
+func (b *GethStatusBackend) SetLogNamespaces(namespaces string) error {
```

Comment:
@osmaczko is there a reason to not expose a data structure based API to clients instead of the string based approach like in `test1.test2:debug,test1.test2.test3:info`?

Reply:
That's a great question. I don't believe having a data structure-based API in this case is feasible, as there are numerous namespaces, sub-namespaces, sub-sub-namespaces, and so on. Moreover, there is no single entity that can know all the registered namespaces. I attempted to address this but eventually stopped, as it would have required significant effort. I want developers (this utility is intended for developers) to have the freedom to configure namespaces however they see fit. I can imagine a developer adjusting namespace settings ad hoc to suit their needs for a specific task or during bug investigations.

As for the Waku debug toggle, I understand it is not ideal, as both clients would need to know the exact namespace. Maybe for that case, `SetWakuLogLevel` could be introduced as a helper, so the knowledge of the namespace is hidden in status-go?

Reply:
Ah okay @osmaczko, you considered this option already 👍🏼 

In any case, I was imagining something simple and easy to serialize, example payload:

```json
{
  "logNamespaces": [
    {"name": "test1.test2",       "level": "debug"},
    {"name": "test1.test2.test3", "level": "info"}
  ]
}
```

Maybe the convenience of using a string vs a data structure is related to the programming language in question? The string approach requires more knowledge about how namespaces are used internally. For example, the dev should know the namespaces are splittable by a comma and that the log level should be separated by a colon. The log level type is a bit more opaque and the dev may need to manually find the regex that validates it deeper down in logutils.

Nothing too important, thanks for considering it.

Reply:
Thanks @ilmotta, that's a nice proposition. 

> Maybe the convenience of using a string vs a data structure is related to the programming language in question?

Not at all. The whole idea of namespaces was inspired by: [go-log](https://github.com/ipfs/go-log/blob/d4825eba3c99b956004c578657a2df9ff08d55f8/README.md?plain=1#L68). The idea behind using a string is to potentially enable configuration through environment variable.

Reply:
> The idea behind using a string is to potentially enable configuration through environment variable.

👍🏼 If namespaces are meant to be _also_ configurable by env vars then that is indeed a good argument in favor of using a string as input.